src/nanoserver/1809/helix/amd64/Dockerfile

```diff
@@ -10,28 +10,31 @@ RUN $apiUrl = 'https://api.github.com/repos/PowerShell/PowerShell/releases/lates
     Expand-Archive -Path C:\powershell.zip -DestinationPath C:\powershell; `
     Remove-Item -Path C:\powershell.zip
 
+# Download latest stable version of Python
+RUN $apiUrl = 'https://api.nuget.org/v3-flatcontainer/python/index.json'; `
+    $response = Invoke-RestMethod -Uri $apiUrl; `
+    $versions = $response.versions | Where-Object { $_ -notmatch '-' } | Sort-Object { [version]$_ } -Descending; `
+    $latestVersion = $versions[0]; `
+    echo \"Downloading Python $latestVersion\"; `
+    Invoke-WebRequest -Uri https://www.nuget.org/api/v2/package/python/$latestVersion -OutFile $env:TEMP\python.zip; `
+    md C:\PythonTemp; `
+    tar -zxf  $env:TEMP\python.zip -C C:\PythonTemp;
+
 
 FROM mcr.microsoft.com/windows/nanoserver:1809
 
-SHELL ["cmd", "/S", "/C"]
 USER ContainerAdministrator
-ENTRYPOINT C:\Windows\System32\cmd.exe
 
 COPY --from=installer [ "C:\\powershell\\", "C:\\Program Files\\PowerShell\\" ]
+COPY --from=installer [ "C:\\PythonTemp\\tools", "C:\\Python\\" ]
 
-RUN curl -SL --output %TEMP%\python.zip https://www.nuget.org/api/v2/package/python/3.7.3 `
-    && md C:\Python C:\PythonTemp `
-    && tar -zxf %TEMP%\python.zip -C C:\PythonTemp `
-    && xcopy /s c:\PythonTemp\tools C:\Python `
-    && rd /s /q c:\PythonTemp `
-    && del /q %TEMP%\python.zip `
-    && setx /M PYTHONPATH "C:\Python\Lib;C:\Python\DLLs;"
+RUN setx /M PATH "%PATH%;C:\Program Files\PowerShell\;C:\Python;C:\python\scripts" && `
+    setx /M PYTHONPATH "C:\Python\Lib;C:\Python\DLLs;" && `
+    setx /M VIRTUAL_ENV "C:\Python-env"
 
-RUN setx /M PATH "%PATH%;C:\Program Files\PowerShell\;C:\Python;C:\python\scripts"
-
-RUN md c:\\helixtmp && pushd c:\\helixtmp &&`
-    python -m pip install --upgrade pip==20.2 && `
-    python -m pip install virtualenv==16.6.0 && `
+RUN md c:\\helixtmp && pushd c:\\helixtmp && `
+    python -m pip install --upgrade pip && `
```

Comment:
please do this upgrade inside the `venv` i.e., move this down one line

Reply:
actually, the container will likely build just fine w/o updating `pip` at all. the upgrade only seems to be necessary when using older Python versions

Reply:
It does more than just upgrades pip. It also ensures that pip is on the `PATH`, for example. So if you take this line out, you can't just run `pip` because it won't be found on the `PATH`.

Reply:
I've updated the pip install to be after venv

Reply:
that's good. however you're not adding `pip.exe` to `$env:PATH` at all. `venv` creation places `python.exe`, `pip.exe`, and so on in `$env:VIRTUAL_ENV\\Scripts`  but I don't see that in `$env:PATH` in the current changes

Reply:
Local testing of removing the pip install shows this output:

```
[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: C:\Python-env\Scripts\python.exe -m pip install --upgrade pip
```

Let me know if the install should be added back because of it.

Reply:
nah, we tend to stick w/ `pip` versions that aren't causing problems or just use what the Python installation provides. can suppress the message w/ `--disable-pip-version-check`. in dotnet-build-machines, we use that on every `python -m pip install ...` command line